src/coord/src/coord.rs

```diff
@@ -954,6 +955,11 @@ impl<S: Append + 'static> Coordinator<S> {
                 );
             }
             Err(e) => {
+                if matches!(e, CoordError::ReadThenWriteTimeoutExceeded(..)) {
```

Comment:
I don't like sniffing for a sentinel here because it feels easy for something to break in the future, and having boolean parameters in functions (cancel_inner here) is a known bad pattern. I've had a variety of ideas that are not good for some reasons, but we need to be able to send a nice error message to the user (not just "peek cancelled"). Is it possible to send a CancelRequest message right above where the ReadThenWriteTimeoutExceeded is created? Or something where we aren't communicating an expected side effect via a specific error message?